__tests__/lib/index.test.js

```diff
@@ -26,42 +21,49 @@ const {
 const debug = debugFactory( 'vip-search-replace:index.test' );
 const processPath = process.cwd();
 
-let readableStream, writeableStream;
-const tmpDir = fs.mkdtempSync( path.join( os.tmpdir(), 'vip-search-replace-tests-' ) );
+let readableStream;
+let writeableStream;
+const tmpDir = mkdtempSync( path.join( tmpdir(), 'vip-search-replace-tests-' ) );
 const readFilePath = path.join( processPath, '__tests__', 'lib', 'in-sample.sql' );
 const writeFilePath = path.join( tmpDir, 'out-sample.sql' );
 const nonZeroExitCodeScript = path.join( processPath, 'bin', 'non-zero-exit-code.sh' );
 
 beforeEach( () => {
-	readableStream = fs.createReadStream( readFilePath );
-	writeableStream = fs.createWriteStream( writeFilePath, { encoding: 'utf8' } );
+	readableStream = createReadStream( readFilePath );
+	writeableStream = createWriteStream( writeFilePath, { encoding: 'utf8' } );
 } );
 
 afterEach( () => {
 	readableStream.close();
 	writeableStream.close();
-	if ( fs.existsSync( writeFilePath ) ) {
-		fs.unlinkSync( writeFilePath );
+	if ( existsSync( writeFilePath ) ) {
+		unlinkSync( writeFilePath );
 	}
 } );
 
+/**
+ * @param {string[]} replacements
+ * @param {string|null} [customScript]
+ * @returns {Promise<{result: Stream, outFile: string | undefined}>}
+ */
 async function testHarness( replacements, customScript = null ) {
 	debug( replacements, customScript );
 	const binary = process.env.CI === 'true' ? './bin/go-search-replace-test' : null;
 	const script = customScript || binary;
 	const result = await replace( readableStream, replacements, script );
-	await new Promise( resolve => {
+
+	await /** @type {Promise<void>} */ ( new Promise( ( resolve ) => {
```

Comment:
This weird syntax is to keep the TS validator happy.

Ref: https://github.com/microsoft/TypeScript/issues/46570#issuecomment-954341021